core/shared/src/main/scala/zio/prelude/These.scala

```diff
@@ -31,6 +31,14 @@ sealed trait These[+A, +B] { self =>
       case (Both(l, r1), Right(r2))     => Both(l, B1.combine(r1, r2))
       case (Both(l1, r1), Both(l2, r2)) => Both(A1.combine(l1, l2), B1.combine(r1, r2))
     }
+
+  def fold[A1 >: A](f: A => A1, g: B => A1)(implicit A1: Associative[A1]): A1 =
```

Comment:
I think the most idiomatic name for this operator is `reduceMap`. This would also let use `fold` for the traditional operator that let's the user specify functions to handle the `left`, `right`, and `both` cases.